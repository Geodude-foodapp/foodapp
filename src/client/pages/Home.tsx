import React, { useState } from 'react';
import RecipeCard from '../components/RecipeCard';

export default () => {
  const [favorites, setFavorites] = useState([1, 2, 3]);
  // const [isUserDataFetched, setUserDataFetched];
  // // TODO: fetch recipe data
  // useEffect(() => {}, []);

  const recipeCards = favorites.map((el) => <RecipeCard key={el} />);

  const getRecipes = (e: React.FormEvent) => {
    e.preventDefault();
    // TODO: hit api for recipe results
  };

  return (
    <section id='home'>
      {/* MAIN DISPLAY */}
      <h1>Home</h1>
      {recipeCards}
    </section>
  );
};
