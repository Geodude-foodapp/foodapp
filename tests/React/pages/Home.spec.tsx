import { render } from '@testing-library/react';
import React from 'react';
import Home from '../../../src/client/pages/Home';

describe('Home Page', () => {
  beforeAll(() => render(<Home />));
  it('matches the snapshot', () => {
    const homePage = document.querySelector('#home');
    expect(homePage).toMatchSnapshot();
  });
});
