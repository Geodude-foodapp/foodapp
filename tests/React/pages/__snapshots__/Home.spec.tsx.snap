// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page matches the snapshot 1`] = `
<section
  id="home"
>
  <h1>
    Home
  </h1>
  <p
    class="recipe-card"
  >
    Recipe Card
  </p>
  <p
    class="recipe-card"
  >
    Recipe Card
  </p>
  <p
    class="recipe-card"
  >
    Recipe Card
  </p>
  <button>
    Sidebar
  </button>
</section>
`;
